lib/src/animated_sampler.dart

```diff
@@ -163,38 +177,50 @@ class _RenderShaderSamplerBuilderWidget extends RenderProxyBox {
     if (size.isEmpty) {
       return;
     }
-
-    if (!enabled) {
-      return super.paint(context, offset);
-    }
-
-    context.pushLayer(
-      _ShaderSamplerBuilderLayer(offset, size, devicePixelRatio, builder),
-      super.paint,
-      offset,
-    );
+    assert(offset == Offset.zero);
+    return super.paint(context, offset);
   }
 }
 
 /// A [Layer] that uses an [AnimatedSamplerBuilder] to create a [ui.Picture]
 /// every time it is added to a scene.
 class _ShaderSamplerBuilderLayer extends OffsetLayer {
-  _ShaderSamplerBuilderLayer(
-    Offset offset,
-    this.size,
-    this.devicePixelRatio,
-    this.callback,
-  ) : super(offset: offset);
+  _ShaderSamplerBuilderLayer(this._callback);
+
+  Size get size => _size;
+  Size _size = Size.zero;
+  set size(Size value) {
+    if (value == size) {
+      return;
+    }
+    _size = value;
+    markNeedsAddToScene();
+  }
 
-  final Size size;
-  final double devicePixelRatio;
-  final AnimatedSamplerBuilder callback;
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio = 1.0;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsAddToScene();
+  }
+
+  AnimatedSamplerBuilder get callback => _callback;
+  AnimatedSamplerBuilder _callback;
+  set callback(AnimatedSamplerBuilder value) {
+    if (value == callback) {
+      return;
+    }
+    _callback = value;
+    markNeedsAddToScene();
+  }
 
   ui.Image _buildChildScene(Rect bounds, double pixelRatio) {
     final ui.SceneBuilder builder = ui.SceneBuilder();
     final Matrix4 transform =
         Matrix4.diagonal3Values(pixelRatio, pixelRatio, 1);
-    transform.translate(-(bounds.left + offset.dx), -(bounds.top + offset.dy));
```

Comment:
We were essentially double transforming this and moving the child canvas offscreen.